rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalDataTransferServer.java

```diff
@@ -0,0 +1,185 @@
+    private static Configuration conf;
+    private static  LocalServer localServer;
+
+    private LocalDataTransferServer()
+    {
+    }
+
+    public static void main(String[] args)
+    {
+        conf = new Configuration();
+        startServer(conf);
+    }
+
+    public static void startServer(Configuration conf)
+    {
+        localServer = new LocalServer(conf);
+        new Thread(localServer).run();
+    }
+
+    public static void stopServer()
+    {
+        if (localServer != null) {
+            localServer.stop();
+        }
+    }
+
+    public static class LocalServer implements Runnable
+    {
+        static ServerSocketChannel listener;
+        Configuration conf;
+
+        public LocalServer(Configuration conf)
+        {
+            this.conf = conf;
+        }
+
+        @Override
+        public void run()
+        {
+            int port = CacheConfig.getLocalServerPort(conf);
+            ExecutorService threadPool = Executors.newCachedThreadPool();
+            try {
+                listener = ServerSocketChannel.open();
+                listener.bind(new InetSocketAddress(port));
+                log.info("Listening on port " + port);
+                while (true) {
+                    SocketChannel clientSocket = listener.accept();
+                    ClientServiceThread cliThread = new ClientServiceThread(clientSocket, conf);
+                    threadPool.execute(cliThread);
+                }
+            }
+            catch (AsynchronousCloseException e) {
+                log.info("Stopping Local Transfer server");
+            }
+            catch (IOException e) {
+                log.error(String.format("Error starting Local Transfer server %s", e));
+            }
+
+        }
+
+        public void stop()
+        {
+            try {
+                listener.close();
+            }
+            catch (IOException e) {
+                log.error(String.format("Error stopping Local Transfer server %s", e));
+            }
+        }
+
+    }
+
+    static class ClientServiceThread
+            extends Thread
+    {
+        SocketChannel localDataTransferClient;
+        RetryingBookkeeperClient bookKeeperClient;
+        Configuration conf;
+
+        ClientServiceThread(SocketChannel s, Configuration conf)
+        {
+            localDataTransferClient = s;
+            this.conf = conf;
+        }
+
+        public void run()
+        {
+            try {
+                log.debug("Connected to node - " + localDataTransferClient.getLocalAddress());
+                BookKeeperFactory bookKeeperFactory = new BookKeeperFactory();
+                ByteBuffer dataInfo = ByteBuffer.allocate(CacheConfig.getMaxHeaderSize(conf));
+
+                int read = localDataTransferClient.read(dataInfo);
+                if (read == -1) {
+                    throw new Exception("Could not read data from Non-local node");
```

Comment:
Log this as error, since this is on a separate thread, it might get lost. Add enough info like client name or ip